packages/core/schematics/migrations/initial-navigation/collector.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+
+import {getImportOfIdentifier} from '../../utils/typescript/imports';
+
+
+/**
+ * Visitor that walks through specified TypeScript nodes and collects all
+ * found RouterModule instances that have `forRoot` call expressions
+ */
+export class Collector {
+  public resolvedModules: ts.CallExpression[] = [];
+
+  constructor(private typeChecker: ts.TypeChecker) {}
+
+  visitNode(node: ts.Node) {
+    if (isRouterModuleCall(this.typeChecker, node)) {
+      this.resolvedModules.push(node);
+    }
+
+    ts.forEachChild(node, n => this.visitNode(n));
+  }
+}
+
+function isRouterModuleCall(typeChecker: ts.TypeChecker, node: ts.Node): node is ts.CallExpression {
+  const moduleName = '@angular/router';
+  const routerModule = 'RouterModule';
+  const functionCall = 'forRoot';
+  if (ts.isCallExpression(node) && ts.isPropertyAccessExpression(node.expression) &&
+      ts.isIdentifier(node.expression.expression)) {
+    const module = node.expression.expression;
+    const call = node.expression.name;
+    const imp = getImportOfIdentifier(typeChecker, module);
```

Comment:
Optional nit for more obvious/correct names:

```suggestion
    const containerExpr = node.expression.expression;
    const functionName = node.expression.name.text;
    const importData = getImportOfIdentifier(typeChecker, module);
```